jwst/datamodels/slit.py

```diff
@@ -0,0 +1,163 @@
+from . import model_base
+from .image import ImageModel
+
+
+__all__ = ['SlitModel', 'SlitDataModel']
+
+
+class SlitDataModel(model_base.DataModel):
+    """
+    A data model for 2D images.
+
+    Parameters
+    ----------
+    init : any
+        Any of the initializers supported by `~jwst.datamodels.DataModel`.
+
+    data : numpy array
+        The science data.
+
+    dq : numpy array
+        The data quality array.
+
+    err : numpy array
+        The error array.
+
+    relsens : numpy array
+        The relative sensitivity table.
+
+    relsens2d: numpy array
+        The relative sensitivty 2D array.
+    """
+
+    schema_url = "slitdata.schema.yaml"
+
+
+    def __init__(self, init=None, data=None, dq=None, err=None,
+                 wavelength=None, var_poisson=None, var_rnoise=None, **kwargs):
+        if isinstance(init, (SlitModel, ImageModel)):
+            super(SlitDataModel, self).__init__(init=None, **kwargs)
+            #self.update(init)
+            self.data = init.data
+            self.dq = init.dq
+            self.err = init.err
+            self.relsens = init.relsens
+            self.area = init.area
+            if hasattr(init, 'wavelength'):
+                self.wavelength = init.wavelength
+            if hasattr(init, 'var_poisson'):
+                self.var_poisson = init.var_poisson
+            if hasattr(init, 'var_rnoise'):
+                self.var_rnoise = init.var_rnoise
+            for key in kwargs:
+                setattr(key, kwargs[key])
+
+            if hasattr(self.meta, 'wcs'):
+                self.meta.wcs = init.meta.wcs
+            else:
+                self.meta.wcs = None
+            return
+        super(SlitDataModel, self).__init__(init=init, **kwargs)
+
+        if data is not None:
+            self.data = data
+
+        if dq is not None:
+            self.dq = dq
+
+        if err is not None:
+            self.err = err
+
```

Comment:
What about area and relsens? Do they need to be in this list?

Reply:
Well, these are created by the photom step but it doesn't hurt to add them here.